QAChat/Data_Processing/pdf_reader.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Emanuel Erben
+# SPDX-FileCopyrightText: 2023 Felix Nützel
+
+
+import io
+from datetime import datetime
+
+from pdfminer.converter import TextConverter
+from pdfminer.layout import LAParams
+from pdfminer.pdfdocument import PDFDocument
+from pdfminer.pdfinterp import PDFResourceManager, PDFPageInterpreter
+from pdfminer.pdfpage import PDFPage
+from pdfminer.pdfparser import PDFParser
+from pdf2image import convert_from_bytes
+import pytesseract
+import os
+import nltk
+from QAChat.Data_Processing.document_embedder import DataInformation, DataSource
+
+TESSDATA_PREFIX = os.getenv('TESSDATA_PREFIX')
+nltk.download('punkt')
+
+
+def read_pdf(pdf_object, before: datetime, after: datetime, data_src: DataSource):
+    # First try to read the PDF with PDFMiner
+    text = __read_pdf_from_object(pdf_object)
```

Comment:
I had expected the function to read the table row by row. Interestingly, however, it seems to read row by row only if there is a blank colomn below (as Online Team Meeting on the screenshot). If there is no blank row, it reads the column until there is a blank column and then moves to the next column (as in the Github rows on the screenshot). Possibly this is relevant, but possibly not.
![image](https://github.com/amosproj/amos2023ss03-qachat/assets/59824388/a95aa3c4-677d-4cc3-bccc-34cccfba4eb1)